product/product.go

```diff
@@ -33,10 +32,32 @@ type Config struct {
 }
 
 type Product struct {
+	l        hclog.Logger
 	Name     string
 	Seekers  []*seeker.Seeker
 	Excludes []string
 	Selects  []string
+	Config   Config
+}
+
+// Run runs the seekers
+func (p *Product) Run() map[string]interface{} {
```

Comment:
I think it's time to make seekers run concurrently on the prod release, but do so with a context for tripping timeouts and storing retry counts. Retries and timeouts are most useful for network requests which might be flaky, and we'd probably want to make it configurable when seekers are custom defined.